storages/csv-storage/src/store_mut.rs

```diff
@@ -0,0 +1,247 @@
+use {
+    crate::{
+        error::{CsvStorageError, ResultExt},
+        CsvStorage,
+    },
+    async_trait::async_trait,
+    csv::Writer,
+    gluesql_core::{
+        data::{Key, Schema},
+        error::Result,
+        store::{DataRow, StoreMut},
+    },
+    std::{
+        collections::BTreeSet,
+        fs::{remove_file, File, OpenOptions},
+        io::Write,
+        {cmp::Ordering, iter::Peekable, vec::IntoIter},
+    },
+};
+
+#[async_trait(?Send)]
+impl StoreMut for CsvStorage {
+    async fn insert_schema(&mut self, schema: &Schema) -> Result<()> {
+        let schema_path = self.schema_path(schema.table_name.as_str());
+        let ddl = schema.to_ddl();
+        let mut file = File::create(schema_path).map_storage_err()?;
+        file.write_all(ddl.as_bytes()).map_storage_err()?;
+
+        let column_defs = match &schema.column_defs {
+            Some(column_defs) => column_defs,
+            None => {
+                return Ok(());
+            }
+        };
+
+        let columns = column_defs
+            .iter()
+            .map(|column_def| column_def.name.as_str())
+            .collect::<Vec<&str>>();
+        let data_path = self.data_path(schema.table_name.as_str());
+
+        File::create(data_path)
+            .map_storage_err()
+            .map(Writer::from_writer)?
+            .write_record(&columns)
+            .map_storage_err()
+    }
+
+    async fn delete_schema(&mut self, table_name: &str) -> Result<()> {
+        let data_path = self.data_path(table_name);
+        if data_path.exists() {
+            remove_file(data_path).map_storage_err()?;
+        }
+
+        let types_path = self.types_path(table_name);
+        if types_path.exists() {
+            remove_file(types_path).map_storage_err()?;
+        }
+
+        let schema_path = self.schema_path(table_name);
+        if schema_path.exists() {
+            remove_file(schema_path).map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn append_data(&mut self, table_name: &str, rows: Vec<DataRow>) -> Result<()> {
+        let (columns, prev_rows) = self.scan_data(table_name)?;
+
+        if columns.is_some() {
+            let data_path = self.data_path(table_name);
+            let mut wtr = OpenOptions::new()
+                .append(true)
+                .open(data_path)
+                .map_storage_err()
+                .map(Writer::from_writer)?;
+
+            for row in rows {
+                let row = convert(row)?;
+
+                wtr.write_record(&row).map_storage_err()?;
+            }
+
+            Ok(())
+        } else {
+            let rows = prev_rows
+                .collect::<Result<Vec<_>>>()?
+                .into_iter()
+                .map(|(_, row)| row)
+                .chain(rows)
+                .collect();
+
+            self.write(table_name, columns, rows)
```

Comment:
How about using mut and extend?
```rs
let mut appended_rows = prev_rows
    .map(|result| Ok(result?.1))
    .collect::<Result<Vec<_>>>()?;

appended_rows.extend(rows);

self.write(table_name, columns, appended_rows)
```

Reply:
how about
```rust
let rows = prev_rows
    .map(|item| item.map(|(_, row)| row))
    .chain(rows.into_iter().map(Ok))
    .collect::<Result<Vec<_>>>()?;
```
this? then we can merge two rows with a single collect.